flink-connector-mysql-cdc/src/main/java/com/ververica/cdc/connectors/mysql/source/reader/MySqlSourceReader.java

```diff
@@ -105,6 +108,8 @@ public void start() {
     protected MySqlSplitState initializedState(MySqlSplit split) {
         if (split.isSnapshotSplit()) {
             return new MySqlSnapshotSplitState(split.asSnapshotSplit());
+        } else if (split.isSuspendSplit()) {
+            return null;
```

Comment:
IIUC `SuspendSplit` is one kind of special split, we also need to snapshot/restore it for consistency